src/weapon/w_033.c

```diff
@@ -3,7 +3,61 @@
 #include "weapon_private.h"
 #include "shared.h"
 
-INCLUDE_ASM("weapon/nonmatchings/w_033", EntityWeaponAttack);
+extern WeaponAnimation D_EB000_8017AC3C[2];
+
+void EntityWeaponAttack(Entity* self) {
+    WeaponAnimation* anim;
+    s8 animIndex;
+
+    self->posX.val = g_Entities->posX.val;
+    self->posY.val = PLAYER.posY.val;
+    self->facingLeft = PLAYER.facingLeft;
+    animIndex = (self->params & 0x7fff) >> 8;
+    anim = &D_EB000_8017AC3C[animIndex];
+
+    if (PLAYER.ext.player.anim >= anim->frameStart &&
+        PLAYER.ext.player.anim < (anim->frameStart + 7) && g_Player.unk46) {
+
+        if (self->step == 0) {
+            SetSpriteBank1(&g_Animset);
```

Comment:
```suggestion
            SetSpriteBank1(g_Animset);
```
it is already a pointer to an array. I am not sure why with `&` matches too :)